pip/tests/test_interpreter.py

```diff
@@ -284,20 +283,31 @@ def test_callables_failing_profile_validation_are_still_registered() -> None:
     assert "Qsc.CapabilitiesCk.UseOfDynamicDouble" in str(excinfo)
     with pytest.raises(Exception) as excinfo:
         e.interpret("Foo()")
-    assert "Qsc.CapabilitiesCk.UseOfDynamicDouble" in str(excinfo)
+    assert "Qsc.Eval.UnboundName" in str(excinfo)
```

Comment:
If I understand correctly, the old behavior is more desirable, but this is the best we can do for now. Could you add a comment explaining why? For the next time we encounter these quirks after encountering a total memory loss? ;)

Reply:
I'll comment some more, yeah. The previous behavior was a bit misleading in the unit test... the error was coming from the definition in the previous interpret call in addition to the call the site, so two errors were being reported. Now it's more consistent with the behavior you'd see when there was an HIR pass failure: the callable is not defined, but the name is still hanging around.